static/js/public/snap-details/channelMap.js

```diff
@@ -1,294 +1,418 @@
-      iframe.style.top = '-9999px';
-      iframe.style.left = '-9999px';
-      iframe.src = `snap://${name}`;
-      document.body.appendChild(iframe);
-
-      if (typeof ga !== 'undefined') {
-        // The first attempt should be counted towards the 'intent'
-        let label = 'Snap install intent';
-        let value = `${name}`;
-
-        // Subsequent attempts should still be tracked, but not as 'intent'
-        if (attempt > 1) {
-          label = 'Snap install click';
-          value += ` - click ${attempt}`;
-        }
+  writeInstallInstructions(channel) {
+    let paramString = '';
 
-        ga('gtm1.send', {
-          hitType: 'event',
-          eventCategory: 'Snap details',
-          eventAction: 'Click view in desktop store button',
-          eventLabel: label,
-          eventValue: value
-        });
+    if (channel.indexOf('latest') === 0) {
+      if (channel.indexOf('stable') !== 7) {
+        paramString = `--${channel.split('/')[1]}`;
       }
-
-      attempt += 1;
+    } else {
+      paramString = `--channel=${channel}`;
     }
-  });
-}
-
-export default function initChannelMap(el, packageName, channelMapData) {
-  initOpenSnapButtons();
 
-  const channelMapEl = document.querySelector(el);
-  const channelOverlayEl = document.querySelector('.p-channel-map-overlay');
+    const template = this.INSTALL_TEMPLATE
+      .split('${channel}')
+      .join(channel)
+      .split('${paramString}')
+      .join(paramString);
+    const holder = document.querySelector('[data-js="channel-map-install-details"]');
 
-  initTabs(channelMapEl);
+    holder.innerHTML = template;
+  }
 
-  let closeTimeout;
+  writeTable(el, data) {
+    let cache;
+    const tbody = data.map((row, i) => {
+      const isSameTrack = (cache && row[0] === cache);
+      let rowClass = [];
 
-  // init open/hide buttons
-  const openChannelMap = (event) => {
-    const openButton = event.target.closest('.js-open-channel-map');
+      if (i === 0) {
+        rowClass.push('is-highlighted');
+      }
 
-    if (openButton) {
-      // open tab based on button click (or install tab by default)
-      const openTabName = openButton.getAttribute('aria-controls') || 'channel-map-tab-install';
+      if (isSameTrack) {
+        rowClass.push('no-border');
+      }
 
-      // clear hiding animation if it's still running
-      clearTimeout(closeTimeout);
+      let _row = `
+<tr class="${rowClass.join(' ')}" data-js="slide-install-instructions" data-channel="${row[0]}/${row[1]}">
+  <td>${row[0]}/${row[1]}</td>
+  <td>${row[2]}</td>
+  <td class="u-hide--medium u-hide--small">${row[3]}</td>
+  <td class="u-align--center"><a href="#" class="p-channel-map__version-table-install">Install &rsaquo;</a></td>
+</tr>`;
+      cache = row[0];
+      return _row;
+    });
+
+    el.innerHTML = tbody.join('');
+  }
 
-      const openTab = channelMapEl.querySelector(`.p-tabs__link[aria-controls=${openTabName}]`);
+  prepareTable(archData) {
```

Comment:
This method contains quite some logic that is hard to follow without better context. I don't fully understand what it does (not knowing what is the shape of the data or what does 'trimmed' tracks mean here). Possibly it would be good to add some comments to give some context on what kind of preparations on the channel map data are done here.